crates/stoffel-vm/src/net/hb_engine.rs

```diff
@@ -0,0 +1,291 @@
+        );
+
+        // Create the MPC node
+        let node = <HoneyBadgerMPCNode<Fr, RBCImpl> as MPCProtocol<
+            Fr,
+            RobustShare<Fr>,
+            QuicNetworkManager,
+        >>::setup(party_id, mpc_opts)
+            .map_err(|e| format!("Failed to create MPC node: {:?}", e))?;
+
+        Ok(Arc::new(Self {
+            instance_id,
+            party_id,
+            n,
+            t,
+            net,
+            node: Arc::new(Mutex::new(node)),
+            ready: AtomicBool::new(false),
+            mul_session_counter: Arc::new(Mutex::new(0)),
+        }))
+    }
+
+    fn ensure_rt() -> Result<tokio::runtime::Handle, String> {
+        tokio::runtime::Handle::try_current().map_err(|e| format!("Tokio runtime not available: {}", e))
+    }
+
+    fn encode_share(share: &RobustShare<Fr>) -> Result<Vec<u8>, String> {
+        let mut out = Vec::new();
+        share.serialize_compressed(&mut out).map_err(|e| format!("serialize share: {}", e))?;
+        Ok(out)
+    }
+
+    fn decode_share(bytes: &[u8]) -> Result<RobustShare<Fr>, String> {
+        RobustShare::<Fr>::deserialize_compressed(bytes).map_err(|e| format!("deserialize share: {}", e))
+    }
+}
+
+impl MpcEngine for HoneyBadgerMpcEngine {
+    fn protocol_name(&self) -> &'static str { "honeybadger-mpc" }
+    fn instance_id(&self) -> u64 { self.instance_id }
+    fn is_ready(&self) -> bool { self.ready.load(Ordering::SeqCst) }
+
+    fn start(&self) -> Result<(), String> {
+        // Mark engine as ready in test/single-process scenarios. Real deployments should call `preprocess()`.
+        self.ready.store(true, Ordering::SeqCst);
+        Ok(())
+    }
+
+    fn input_share(&self, ty: ShareType, clear: &Value) -> Result<Vec<u8>, String> {
+        // Minimal support: ShareType::Int over Fr via direct embedding (u64 -> Fr).
+        match (ty, clear) {
+            (ShareType::Int(_), Value::I64(v)) => {
+                let secret = Fr::from(*v as u64);
+                let mut rng = ark_std::test_rng();
+                let shares = RobustShare::compute_shares(secret, self.n, self.t, None, &mut rng)
+                    .map_err(|e| format!("compute_shares: {:?}", e))?;
+                let my = &shares[self.party_id];
+                Self::encode_share(my)
+            }
+            (ShareType::Bool(_), Value::Bool(b)) => {
+                let secret = if *b { Fr::from(1u64) } else { Fr::from(0u64) };
+                let mut rng = ark_std::test_rng();
+                let shares = RobustShare::compute_shares(secret, self.n, self.t, None, &mut rng)
+                    .map_err(|e| format!("compute_shares: {:?}", e))?;
+                let my = &shares[self.party_id];
+                Self::encode_share(my)
+            }
+            (ShareType::Float(_), Value::Float(fp)) => {
+                // Basic fixed-point: assume input already scaled as i64; map to field via u64 cast.
+                let secret = Fr::from(*fp as u64);
```

Comment:
By casting an i64 to a u64, are we sure we are not losing information here? What if something was incorrectly a negative i64 but then gets casted to a u64 resulting in an unintended result?

Reply:
Tbh this is placeholder since we dont have floating point yet in mpc-protocols in a state that lets me implement in the VM

Reply:
Ah, that's in a PR at the moment. Should be ready for an integration by next week